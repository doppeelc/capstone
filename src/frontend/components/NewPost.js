import React, { useState, useContext } from "react";
import UserContext from "./UserContext";
import "../styles/NewPost.css";


function NewPost() {
    const { currentUser, number, advice, chuck, makePost } = useContext(UserContext);

    const [formData, setFormData] = useState({
        fact:""
    });

    if(!currentUser.loaded) {
        return <p>Loading...</p>
    }

    async function doNumber() {
        let fact = await number.getNumberFact();
        setFormData({fact});
    }

    async function doAdvice() {
        let fact = await advice.getAdvice();
        setFormData({fact});
    }

    async function doChuck() {
        let fact = await chuck.getChuckJoke();
        setFormData({fact});
    }

    async function postPost(e) {
        e.preventDefault();
        await makePost.makePost(formData.fact);
        setFormData({fact:""});
        window.location.href = "/profile";
    }


    return (
        <div id="NewPost" >
            <div id="NewPost-options" >
                <button id="NewPost-button" onClick={doNumber} >Number Fact</button>
                <button id="NewPost-button" onClick={doAdvice} >Life Advice</button>
                <button id="NewPost-button" onClick={doChuck} >Chuck Norris Joke</button>
            </div>
            <form id="NewPost-form" onSubmit={postPost} >
                <p id="NewPost-post" >{formData.fact}</p>
                <button id="NewPost-formButton" >Post</button>
            </form>
        </div>
    )
}

export default NewPost;