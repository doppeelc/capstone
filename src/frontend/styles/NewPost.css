#NewPost {
    margin: 25px;
    background-color: lightgray;
}

#NewPost-form {
    height: 80px;
}

#NewPost-post {
    margin-top: 15px;
    height: 30px;
}

#NewPost-button {
    margin-top: 15px;
}