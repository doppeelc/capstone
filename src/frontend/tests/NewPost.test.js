import { render } from '@testing-library/react';
import NewPost from '../components/NewPost';
import UserContext from '../components/UserContext';
import { currentUser } from './__testCommon';

test('renders without crashing', () => {
    render(
        <UserContext.Provider value={{currentUser:currentUser}}>
            <NewPost />
        </UserContext.Provider>
    );
});

test('matches the snapshot', () => {
    const { asFragment } = render(
        <UserContext.Provider value={{currentUser:currentUser}}>
            <NewPost />
        </UserContext.Provider>
    );
    expect(asFragment()).toMatchSnapshot();
});






