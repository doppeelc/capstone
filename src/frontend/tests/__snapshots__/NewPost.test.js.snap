// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    id="NewPost"
  >
    <div
      id="NewPost-options"
    >
      <button
        id="NewPost-button"
      >
        Number Fact
      </button>
      <button
        id="NewPost-button"
      >
        Life Advice
      </button>
      <button
        id="NewPost-button"
      >
        Chuck Norris Joke
      </button>
    </div>
    <form
      id="NewPost-form"
    >
      <p
        id="NewPost-post"
      />
      <button
        id="NewPost-formButton"
      >
        Post
      </button>
    </form>
  </div>
</DocumentFragment>
`;
